lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -199,7 +205,8 @@ export abstract class ClientApplication {
      */
     async acquireTokenRedirect(request: RedirectRequest): Promise<void> {
         // Preflight request
-        this.logger.verbose("acquireTokenRedirect called");
+        request.correlationId = this.getRequestCorrelationId(request);
```

Comment:
We shouldn't directly change the request. This can result in subsequent requests using the same correlationId

Reply:
Fixed and added unit tests.